lib/matplotlib/backends/backend_qt.py

```diff
@@ -642,8 +642,13 @@ def __init__(self, canvas, parent=None, coordinates=True):
             if text is None:
                 self.addSeparator()
             else:
+                slot = getattr(self, callback)
+                # https://bugreports.qt.io/browse/PYSIDE-2512
+                slot = functools.wraps(slot)(functools.partial(slot))
+                slot = QtCore.Slot()(slot)
```

Comment:
If this indeed portable this way, can we just decorate the methods directly?  Even if we have to add a bunch of methods like

```python
@QtCore.Slot
def foo(self):
    return super().foo()
```

that seems better than doing the multi-layered meta-programming here!

Reply:
Yes, can do. A considerable amount of meta-programming is still done in the loop just around the snippet, so maybe a test is needed to make sure that all functions in `self.toolitems` have been decorated...

Reply:
Wow, that's interesting. I tried adding 

```python
    @QtCore.Slot()
    def home(self, *args):
        return super().home(*args)

    @QtCore.Slot()
    def back(self, *args):
        return super().back(*args)

    @QtCore.Slot()
    def forward(self, *args):
        return super().forward(*args)

    @QtCore.Slot()
    def pan(self, *args):
        return super().pan(*args)

    @QtCore.Slot()
    def zoom(self, *args):
        return super().zoom(*args)

    @QtCore.Slot()
    def configure_subplots(self, *args):
        return super().configure_subplots(*args)

    @QtCore.Slot()
    def edit_parameters(self, *args):
        return super().edit_parameters(*args)

    @QtCore.Slot()
    def save_figure(self, *args):
        return super().save_figure(*args)
```

and this works for `home`, `back` and `forward` - but not for the others. Irrespective of the position inside `NavigationToolbar2QT`, and irrespective of the order of the functions. This is with `PySide6-Essentials` v6.6.0.

I put this code into https://github.com/bersbersbers/matplotlib-27214/commit/a3d97f8eb2b40a46fe2d4fc4cf60f9d631fd3cf2 if anyone wants to try it themselves.

Reply:
My guess is multiple inheritance is related, but do not have a good theory as to why (other than general suspicion that MI always causes problems....).

Reply:
If the metaprogramming approach works then let's just stick to it.